Adds a new traitor item to the emagged coffee vending machine.

Starlight991 opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->
Emagging a coffee vending machine will give out the option to dispense molten chocolate. Molten chocolate is a powerful item that heavily burns and increases the temperature of anyone it gets splashed on.
## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
Would be a difficult but rewarding single target weapon. It only has a chance to be in stock in the emagged machine and the machines are usually in public areas therefor making it risky. also you only get 20u which is just under enough to kill someone. 
## Technical details
<!-- Summary of code changes for easier review. -->
its all yaml from me, a yaml warrior
## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->

https://github.com/user-attachments/assets/5c88603b-7328-4558-9207-96f99d31f5a9

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->
n/a
**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->

:cl:
- add: When emagged, the Solar's Hot Drinks machine gains a secret new drink

beck-thompson commented on Resources/Locale/en-US/reagents/generic.ftl:
```diff
@@ -9,3 +9,4 @@ generic-reagent-effect-parched = You feel parched.
 generic-reagent-effect-thirsty = You feel thirsty.
 generic-reagent-effect-sick = You feel sick after consuming that...
 generic-reagent-effect-slicing-insides = You feel an incredibly sharp pain in your gut!
+generic-reagent-effect-melting-insides = Your insides feel like they are melting!
```
you need a newline at the end of the file

beck-thompson commented on Resources/Locale/en-US/reagents/meta/consumable/drink/drinks.ftl:
```diff
@@ -93,3 +93,6 @@ reagent-desc-posca = Poor warriors' drink from a forgotten era.
 reagent-desc-mopwata = Dirty, stagnant mop water.
+
+reagent-name-molten-chocolate = molten chocolate
+reagent-desc-molten-chocolate = A cup of hot chocolate mixed with a unknown chemical that has been superheated through unknown means.
```
same here

beck-thompson reviewed (changes requested):
Currently emag interactions are frozen! Could you make this obtainable in another way?

Entvari commented:
Damage is **way** too high.
Allows you to near instantly crit any target with ease at a measly TC cost

Entvari commented:
> pen bomb
Not starting a balance argument, but pen bomb costs nearly the same as an Emag, is completely onetime use, is far less effective against armored targets, requires you to somewhat position, doesn't come with a ton of other stuff, etc, etc.
40 heat per swing + extra DoT is actually unreal, especially considering the fact that even if you miss you can just swing again with no penalty. 
This would be a lot more acceptable if it were a onetime purchase but it'd be pretty easy to get multiple of these as most stations have several coffee vendors, this paired with the fact you ALSO get all the benefits of the Emag at the same cost is too much.

Starlight991 commented:
Updated to no longer be obtainable from an emagged vending machine. Now craftable with chemistry using hot chocolate, vestine, plasma and heating. Fixed the newline issues as well.

Geoengi commented:
Now only if we could make putty copies of sec's keycards...

deltanedas commented:
and use duct tape to cuff people

beck-thompson reviewed (approved):
I'm unsure about balance and stuff but it looks fine!
